### ISSUE TITLE
Add how to configure your editor section

### ISSUE THREAD

**Issue by chrisburr** [hsf-training/analysis-essentials#2]
Imported from https://github.com/lhcb/analysis-essentials/issues/50.

I think we should have a section dedicated to how to configure popular text editors with a minimal configuration as easily as possible. This should be pushed in the prerequisites and only mentioned during lessons. Tutorials and customised configuration can be provided as links.

Obviously the definition of “minimal configuration” is debatable but my proposal is:
- [ ] Only use spaces and not tabs
- [ ] Make whitespace visible
- [ ] Inline linting for Python

Editors we should provide settings for:
- emacs @roelaaij 
- vs code
- vim
- sublime @chrisburr
- maybe atom?

**Comment by chrisburr** [comment]
From @eduardo-rodrigues in [this comment](https://github.com/lhcb/analysis-essentials/issues/50#issuecomment-403280331):

Sounds good to me. Definitely settings for Atom are relevant these days ;-).

Personally I use Atom more and more, though I still have to learn fancy capabilities. I use the
Nord theme, see https://atom.io/themes/nord-atom-ui.

**Comment by dberzano** [comment]
Something that might be useful to us (at least that's what I do with Vscode and vim) is explaining how to configure some "actions" you can run from an editor.

* I use [Syntastic](https://github.com/vim-syntastic/syntastic) in vim which works reasonably well when I develop Python
* I configure some default build actions in Vscode in order to build my C++ projects with a keystroke

I fully agree with the following statement from @chrisburr:

> This should be pushed in the prerequisites and only mentioned during lessons.

I can help with the two editors above.

I would also add Xcode to the list (some people use it) and it's also worth reminding that Vscode saves some hidden files inside your current repository, therefore this should be handled appropriately by your `.gitignore`.

**Comment by chrisburr** [comment]
I agree that it would be nice to add but possibly as a separate tutorial. I think it is important that these editor tutorials are short and reliable in the hope it that they are more widely adopted.

*Dumping this here as an idea:* How about we have "cheatsheets" in another repository like:
 - Set up `(vim|emacs|sublime|atom|.*)` with a minimal configuration
 - Set up more advanced things in your editor
 - Minimal git (create `.gitignore`, set remote and run `git status`, `git commit -m "...", `git push`)
